### ISSUE TITLE
Add remapping support for Imports

### ISSUE THREAD

**Issue by crisgarner** [intellij-solidity/intellij-solidity#281]
Popular development frameworks like Dapp.tools and Foundry have the ability to import libraries using remappings, allowing to have a more friendly support for imports (for example open zeppelin, chainlink or ds-test libraries).
 
![Screenshot from 2022-01-03 13-27-55](https://user-images.githubusercontent.com/578688/147971603-801b4401-4fe8-4a48-aec5-5a7bcea939be.png)

The problem is that there is no way to specify the remmapings using intellij-solidity plugin, which leads to import errors on the IDE

![Screenshot from 2022-01-03 13-28-34](https://user-images.githubusercontent.com/578688/147971671-c506de22-4e1c-4dcd-b3f6-87bb145a6753.png)

**Comment by mason** [comment]
+1 to this. also imports like these can't be located (@openzeppelin) without remapping them to a directory that contains the solidity files
```
import {
    SafeERC20,
    SafeMath,
    IERC20,
    Address
} from "@openzeppelin/contracts/token/ERC20/SafeERC20.sol";
```

**Comment by kovalevvlad** [comment]
same here. OpenZeppelin imports are marked as missing

**Comment by jpiabrantes** [comment]
I see this was merged, but I still can't get `remappings.txt` to work. I am on version 2.3.9.

**Comment by abancu** [comment]
this is because that was not implemented, it is work in progress, hopefully, I am also looking for it but I can't see any other activity towards this in the WIP fork... https://github.com/dyzz/intellij-solidity/commits/master

**Comment by aashborn** [comment]
I also use Foundry and I also have the problem with import navigation.

**Comment by aminlatifi** [comment]
@dyzz This feature will be a great help for devs using foundry. Do you have any estimate for its release time?

**Comment by dyzz** [comment]
> this is because that was not implemented, it is work in progress, hopefully, I am also looking for it but I can't see any other activity towards this in the WIP fork... https://github.com/dyzz/intellij-solidity/commits/master


> @dyzz This feature will be a great help for devs using foundry. Do you have any estimate for its release time?

I will try to finish the support for remappings.txt. I'm also facing navigation issues every day.

**Comment by androolloyd** [comment]
Any update on this, happy to kick in some ETH towards a bounty.

**Comment by dyzz** [comment]
finally got some time to work on this. 
One dirty hack to my mind is adapting to different directory schemes of popular projects. e.g. searching code from both "/src/" and "/contracts/". This might helps to work out-of-box without "remapping.txt".
The goal is still to fully support "remapping.txt".

**Comment by androolloyd** [comment]
I think the core part of the issue is that we can't adopt our own import flow like for instance we remap ./src to core so that our imports many folders deep are clean but no navigation makes a same man cry.

**Comment by 0x00ga** [comment]
working for anyone?

**Comment by androolloyd** [comment]
not sure if its been published in the latest version or not yet, I ended up building it on my own to get it working.

**Comment by dyzz** [comment]
> 
It has been released yet. If you have built your own version, can you give an example of this? The current logic is replacing the path using remappings.txt.

**Comment by dyzz** [comment]
> working for anyone?
> 
> I have remappings.txt at root directory, yet all imports are broken. Works like a charm on vscode

It has been released yet. If you have built your own version, can you give an example of this? The current logic is replacing the path using remappings.txt.

**Comment by androolloyd** [comment]
i mean that I just built the plugin from source.

**Comment by 0x00ga** [comment]
> It has been released yet. If you have built your own version, can you give an example of this? The current logic is replacing the path using remappings.txt.

Yeah, it didn't work inmediately after building and importing 3.11, but worked for some reason after reinstalling. Ignore this.

**Comment by alex-grover** [comment]
@dyzz would it be possible to add support for reading the `remappings` key from `foundry.toml` in addition to `remappings.txt`?

**Comment by dyzz** [comment]
> @dyzz would it be possible to add support for reading the `remappings` key from `foundry.toml` in addition to `remappings.txt`?

The goal is to make it an out-of-the-box expirence. I'll look into making this change, but not sure about the schedule for now. As a workaround, you can use the `forge remappings > remappings.txt`

**Comment by seunlanlege** [comment]
This still doesn't work unfortunately v2.4.4

**Comment by SerCeMan** [comment]
👋 Thank you to amazing contributions, I believe remapping has been implemented for `foundry.toml` (https://github.com/intellij-solidity/intellij-solidity/pull/452) and `remappings .txt` (https://github.com/intellij-solidity/intellij-solidity/pull/331). So I'll close this issue. 

If you're still experiencing some issues, please open a new issue, and we'll have a look!
